src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,26 +1461,115 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
+                                    {
+                                        this.StateWhenFalse[slot] = true;
+                                    }
+                                    else
+                                    {
+                                        this.StateWhenTrue[slot] = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            ImmutableArray<int> getOperandSlots(BoundExpression operand)
+            {
+                Debug.Assert(operand != null);
+                Debug.Assert(_lastConditionalAccessSlot == -1);
+                ArrayBuilder<int> slotListBuilder = null;
+                int slot;
 
+                do
+                {
+                    // Due to the nature of binding, if there are conditional access they will be at the top of the bound tree,
+                    // potentially with a conversion on top of it. We go through any conditional accesses, adding slots for the
+                    // condtional receivers if they have them. If we ever get to a receiver that MakeSlot doesn't return a slot
+                    // for, nothing underneath is trackable and we bail at that point. Example:
+                    //
+                    //     a?.GetB()?.C // a is a field, GetB is a method, and C is a property
+                    //
+                    // The top of the tree is the a?.GetB() conditional call. We'll ask for a slot for a, and we'll get one because
+                    // locals have slots. The AccessExpression of the BoundCondtionalAccess is another BoundCondtionalAccess, this time
+                    // with a reciever of the GetB() BoundCall. Attempting to get a slot for this receiver will fail, and we'll
+                    // return an array with just the slot for a.
+                    switch (operand.Kind)
+                    {
+                        case BoundKind.Conversion:
+                            operand = ((BoundConversion)operand).Operand;
+                            continue;
+                        case BoundKind.ConditionalAccess:
+                            var conditional = (BoundConditionalAccess)operand;
+
+                            slot = MakeSlot(conditional.Receiver);
                             if (slot > 0)
                             {
-                                if (slot >= this.State.Capacity) Normalize(ref this.State);
-
-                                Split();
+                                // If we got a slot we must have processed the previous conditional receiver.
+                                Debug.Assert(_lastConditionalAccessSlot == -1);
 
-                                if (op == BinaryOperatorKind.Equal)
+                                if (slot >= this.State.Capacity) Normalize(ref this.State);
+                                if (slotListBuilder == null)
```

Comment:
Let's avoid creating the builder in several places. Consider creating upfront in the caller.